R/module_init_data.R

```diff
@@ -56,7 +56,7 @@ ui_init_data <- function(id, data) {
 #' @rdname module_init_data
 srv_init_data <- function(id, data, modules, filter = teal_slices()) {
   checkmate::assert_character(id, max.len = 1, any.missing = FALSE)
-  checkmate::assert_multi_class(data, c("teal_data", "teal_data_module", "reactive", "reactiveVal"))
+  checkmate::assert_multi_class(data, c("teal_data", "qenv.error", "teal_data_module", "reactive", "reactiveVal"))
```

Comment:
I think providing `qenv.error` to the `data` and let the app open is wrong.

```suggestion
  checkmate::assert_multi_class(data, c("teal_data", "teal_data_module", "reactive", "reactiveVal"))
```

Reply:
This was done to make sure that we show the error message when the input `teal_data` has error when used inside a teal module.
The alternative was a shiny disconnect error where the R console error would point to the app code where the error is. I preferred showing the error message in the app since the app has already started.
Do you think we should revert this change and have the previous disconnection behavior?

```
# 3. Teal Module with teal_data
modules <- modules(example_module(), example_module("mtcars only", datanames = "mtcars"))
ui <- fluidPage(
  "Custom UI",
  ui_teal(id = "teal_1", modules = modules),
  ui_teal(id = "teal_2", modules = modules)
)
server <- function(input, output, session) {
  data <- teal_data(iris = iris, mtcars = mtcars)
  srv_teal(
    id = "teal_1",
    data = within(teal_data(), {
      stop("error")
    }),
    modules = modules
  )
  srv_teal(
    id = "teal_2",
    data = within(teal_data(), {
      stop("error")
    }),
    modules = modules
  )
}
shinyApp(ui, server)
```

Reply:
Yeah, good point 👍  `teal::init` will still throw and app won't be built.